### PR TITLE
try pools with price>0 first

### PR DESCRIPTION
**Pull request by alexcos20**

### CODE REVIEW

**Comment by alexcos20** [comment]
already deployed on mainnet and it's working fine

File: aquarius/events/metadata_updater.py [hunk 1]
```diff
@@ -245,10 +252,12 @@ def get_dt_addresses_from_pool_logs(self, from_block, to_block=None):
...
                 for l in logs:
                     parsed_log = get_event_data(event_abis[i], l)
-                    addresses.extend([(parsed_log.args.get(arg, ''), parsed_log.address) for arg in args])
+                    addresses.extend(
+                        [(parsed_log.args.get(arg, ''), parsed_log.address) for arg in args])
                     # all_logs.append(parsed_log)
 
-        addresses_and_pools = [(a, pool) if a and a.lower() != self._OCEAN else ('', pool) for (a, pool) in addresses]
+        addresses_and_pools = [(a, pool) if a and a.lower(
+        ) != self._OCEAN else ('', pool) for (a, pool) in addresses]
```
Comment (ssallam) [inline aquarius/events/metadata_updater.py:260]: indentation missing, breaking at `else` or `for` makes it easier to read

File: aquarius/events/metadata_updater.py [hunk 2]
```diff
@@ -268,13 +277,16 @@ def get_datatoken_pools(self, dt_address, from_block=0, to_block='latest'):
             return None
 
         pools = [get_event_data(event_abi, log).address for log in logs]
```
Comment (ssallam) [inline aquarius/events/metadata_updater.py:279]: could turn this into a set comprehension by replacing `[ ]` with `{ }`:
```
pools = {get_event_data(event_abi, log).address for log in logs}
```

File: aquarius/events/metadata_updater.py [hunk 3]
```diff
@@ -268,13 +277,16 @@ def get_datatoken_pools(self, dt_address, from_block=0, to_block='latest'):
...
-        return pools
+        return list(set(pools))
```
Comment (ssallam) [inline aquarius/events/metadata_updater.py:280]: remove `set` after making the change above.

File: aquarius/events/metadata_updater.py [hunk 4]
```diff
@@ -268,13 +277,16 @@ def get_datatoken_pools(self, dt_address, from_block=0, to_block='latest'):
...
 
     def _get_liquidity_and_price(self, pools, dt_address):
         assert pools, f'pools should not be empty, got {pools}'
+        logger.debug(f' Searching {pools} for {dt_address}')
         dt_address_lower = dt_address.lower()
+        ocean_address_lower = self._OCEAN.lower()
```
Comment (ssallam) [inline aquarius/events/metadata_updater.py:286]: self._OCEAN is already lower case, to make this more clear, rename `self._OCEAN` to `self._ocean_lower_case` or so.

File: aquarius/events/metadata_updater.py [hunk 5]
```diff
@@ -268,13 +277,16 @@ def get_datatoken_pools(self, dt_address, from_block=0, to_block='latest'):
...
         pool_to_price = dict()
         for _pool in pools:
+            logger.debug(f' Searching {_pool}')
             try:
                 pool = BPool(_pool)
                 pool.getCurrentTokens()
```
Comment (ssallam) [inline aquarius/events/metadata_updater.py:292]: Unused call, should be removed.

File: aquarius/events/metadata_updater.py [hunk 6]
```diff
@@ -283,50 +295,60 @@ def _get_liquidity_and_price(self, pools, dt_address):
...
+                
                 pool_to_price[_pool] = price
+                logger.debug(f' Adding pool {_pool} with price {price}')
 
             except Exception as e:
-                logger.error(f'failed to get liquidity/price info from pool {_pool} and datatoken {dt_address}')
-                #return 0.0, 0.0, 0.0, _pool
+                logger.error(
+                    f'failed to get liquidity/price info from pool {_pool} and datatoken {dt_address}')
+                # return 0.0, 0.0, 0.0, _pool
```
Comment (ssallam) [inline aquarius/events/metadata_updater.py:314]: commented `return` should be removed.

File: aquarius/events/metadata_updater.py (hunk 2 above)
Comment (alexcos20) [reply aquarius/events/metadata_updater.py:278]: no, that is not going to remove duplicates.
but doing [] and then list(set([])) will do that :)

File: aquarius/events/metadata_updater.py (hunk 3 above)
Comment (alexcos20) [reply aquarius/events/metadata_updater.py:279]: see above

File: aquarius/events/metadata_updater.py (hunk 2 above)
Comment (ssallam) [reply aquarius/events/metadata_updater.py:278]: set comprehension will remove duplicates, try it in a python console, it's the same as doing a set([]) but faster and shorter.

**Review by ssallam — APPROVED** [review]

**Pull request merged**